common.gypi

```diff
@@ -367,7 +367,6 @@
         'ldflags': [ '-pthread' ],
       }],
       [ 'OS in "linux freebsd openbsd solaris android aix cloudabi"', {
-        'cflags': [ '-Wall', '-Wextra', '-Wno-unused-parameter', ],
```

Comment:
Does this disable the warnings by default for addons? That would be pretty bad…

Reply:
It doesn't exactly disable, it "un-enables", but yes it changes default behaviour.
This does bring up the issue of `common.gypi` not exactly being the right fix for addons (https://github.com/nodejs/node-gyp/issues/1118)

Authors could set their own warning level in the addons `binding.gyp`.

If you think this should not change for addons, I could try to figure out another mechanism.

Reply:
Are you okay with leaving that out of this PR? We can revisit it once something has happened along the lines of the PR you linked…

Reply:
Ok.